python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -867,6 +879,8 @@ def submit_run_request(run_request) -> SubmitRunRequestResult:
         if isinstance(run, SkippedSensorRun):
             skipped_runs.append(run)
             context.add_run_info(run_id=None, run_key=run_request_result.run_key)
+        elif isinstance(run, BackfillRun):
+            context.add_run_info(run_id=run.backfill_id)
```

Comment:
hm recognizing that this predates this PR - how does one know looking a tick if the 'run_id' field is a run or a backfill?

I think if i could go back in time I would recommend a separate 'backfill_id' field instead of overriding 'run_id'

We could consider renaming add_run_info to like... add_submission_info

Reply:
The omission of run_key is intentional here I assume, backfills can't include run keys right?

Reply:
Regular run ids and backfill ids have different formats (uuid4 vs a string of random lowercase letters), so we're technically able to just parse out which one's which by checking the length or using a regex: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/utils.py?L81

Re: creating a separate `backfill_id`, @jamiedemaria can probably speak to future plans better than I can, but I think the hope is that at some point in the medium future we will no longer have "backfills" as a separate concept (just runs which spawn other runs), and so having a single run_id field at that point in time will be correct. So it's a matter of just living with the weird double purpose of this field until that point in time, rather than forever.

Re: omitting run_key, that's correct yep -- there will never be a run key for a backfill at the moment.

Reply:
future plans - the goal is to eliminate backfills entirely and execute them as runs. How exactly that will be done (whether it's runs that spawn other runs, or runs that behave more like runs with dynamic outputs) is still fuzzy.

Reply:
> I think if i could go back in time I would recommend a separate 'backfill_id' field instead of overriding 'run_id'

i dont believe anything has been built on top of DA emitting backfills yet, and no one is emitting backfills from DA yet either. So if there is compelling reason to switch to having a `backfill_id` field, I think there's still time to make that change without any consequences